packages/mds-compliance-engine/@types/index.ts

```diff
@@ -0,0 +1,36 @@
+import { Device, Telemetry, Timestamp, UUID, VehicleEvent, VEHICLE_EVENT, VEHICLE_STATE } from '@mds-core/mds-types'
+
+export type VehicleEventWithTelemetry = VehicleEvent & { telemetry: Telemetry }
+export type MatchedVehicleWithRule = { [d: string]: { device: Device; rule_applied?: UUID; rules_matched?: UUID[] } }
+
+export interface MatchedVehicleInformation {
+  device_id: UUID
+  state: VEHICLE_STATE
+  event_types: VEHICLE_EVENT[]
+  timestamp: Timestamp
+  /* Sometimes a device can match more than one rule, and it's helpful to know all of them,
+     for instance, with a count policy.
+  */
+  rules_matched: UUID[]
+  rule_applied?: UUID | null // a device can only ever match one rule at most for the purpose of computing compliance, however
+  speed?: number | null
```

Comment:
this is me being a little slow with ts but -- isn't the `?` sufficient to not include `| null` ?

Reply:
Nah, this gives it the option to be `undefined` as well.

Reply:
Do we need to have this both be undefinable *and* nullable? I generally try to lean on nullable whenever I want to explicitly omit something.

Reply:
I didn't want to have to explicitly set `speed` to null on vehicles that aren't being processed by speed type policies. Speaking of which, this is reminding me that I forgot to set the speed on matched vehicles in the speed compliance response.

Reply:
Then why not just use `undefined`? I guess I don't understand why it needs to be both possibly null and possibly undefined, is it an artifact from how mds-agency treats telemetry?

Reply:
Changing it to remove `null` in this type.